docs/fxa-engineering/fxa-webchannel-protocol.md

```diff
@@ -0,0 +1,324 @@
+
+Sent on signin or signup after a user has entered their credentials and submit the form. A response with an `ok` field is expected.
+
+##### request data
+
+```js
+{
+  email: <email>
+}
+```
+
+##### response data
+
+```js
+{
+  ok: true | false;
+}
+```
+
+#### fxaccounts:login
+
+Sent when a user successfully authenticates with Firefox Accounts and sync can begin. This includes after a user signs up but before they verify their email address, after an account unlock, and after a password reset. No response is expected.
+
+##### data
+
+See [Login Data](#logindata).
+
+#### fxaccounts:oauth_login
+
+Sent when a user successfully authenticates via a WebChannel OAuth flow.
+
+##### data
+
+```
+{
+  "code": "02f3cfea84ac4c143662b38d6c7f0c82c6f91eb041befc7cecda446b1b4887c1",
+  "state": "vHao1p6OizzwReCkQMSpZA",
+  "redirect": "urn:ietf:wg:oauth:2.0:oob:oauth-redirect-webchannel",
+  "action": "signin",
+  "declinedSyncEngines": ["history"],
+  "offeredSyncEngines": ["history"]
+}
+```
+
+##### support
+
+Supported by `oauth_webchannel_v1` context.
+
+#### fxaccounts:verified
+
+The user has successfully verified their email address.
+
+##### data
+
+See [Login Data](#logindata).
+
+##### support
+
+#### fxaccounts:delete_account
+
+Sent after the user successfully deletes their account. No response is expected.
+Expected by Fennec.
+
+##### data
+
+```js
+{
+  email: <email>,
+  uid: <user id>
+}
+```
+
+- `email` - User's email address.
+- `uid` - The user identifier.
+
+#### fxaccounts:delete
+
+Sent after the user successfully deletes their account. No response is expected.
+Expected by Fx Desktop.
+
+##### data
+
+```js
+{
+  email: <email>,
+  uid: <user id>
+}
+```
+
+- `email` - User's email address.
+- `uid` - The user identifier.
+
+#### fxaccount:change_password
+
+Sent after the user successfully changes their password. No response is expected. A change password causes a user's `unwrapBKey` to change, all data that is sent with `fxaccounts:login` is sent with `fxaccounts:change_password`.
+
+##### data
+
+See [Login Data](#logindata).
+
+#### fxaccounts:logout
+
+Called whenever the user signs out of their account. No response is expected.
+
+##### data
+
+```js
+{
+  uid: <user id>
+}
+```
+
+- `uid` - The user identifier.
+
+#### profile:change
+
+Sent after the user changes their avatar or display name. No response is
+expected.
+
+##### data
+
+```js
+{
+  uid: <user id>
+}
+```
+
+- `uid` - The user identifier.
+
+### Command data values
+
+#### loginData
+
+```js
+{
+  customizeSync: (true|false),
+  declinedSyncEngines: [<array of declined sync engines>],
+  email: <email>,
+  keyFetchToken: <key fetch token>,
+  offeredSyncEngines: [<array of displayed sync engines>],
+  sessionToken: <session token>,
+  uid: <user id>,
+  unwrapBKey: <unwrap b key>,
+  verified: (true|false),
+  verifiedCanLinkAccount: (true|false)
+}
+```
+
+- `customizeSync` - Should the user be shown the browser's `Customize Sync` dialog?
```

Comment:
We should not that this is deprecated.

Reply:
Right, taking it out of the doc. We totally removed the capability in https://github.com/mozilla/fxa/commit/52c86a8d1.